File: dSoak-Common-Java/src/SharedObject/ProcessData.java
package SharedObject;

import com.google.gson.annotations.Expose;

public class ProcessData 
{
	@Expose public short GameId;
	@Expose public short ProcessId;
	@Expose public PossibleProcessType ProcessType; 
	@Expose public short LifePoints; 
	@Expose public short HitPoints; 
	@Expose public short NumberOfPennies; 
	@Expose public short NumberOfUnfilledBalloon; 
	@Expose public short NumberOfFilledBalloon; 
	@Expose public short NumberOfUnraisedUmbrellas; 
	@Expose public boolean HasUmbrellaRaised; 
    
    public enum PossibleProcessType
	{ 
		Unknown((short)0),
		Player((short)1), 
		BalloonStore((short)2), 
		WaterServer((short)3), 
		UmbrellaSupplier((short)4), 
		Thief((short)5); 
	
	
		short value;
		PossibleProcessType(short va) { this.value = va; }
		
		public short getValue()
		{
			return value;
		}
	};
}

File: dSoak-Common-CSharp/SharedObjects/ProcessData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Runtime.Serialization;

namespace SharedObjects
{
    [DataContract]
    public class ProcessData
    {
        public enum PossibleProcessType { Unknown = 0, Player = 1, BalloonStore = 2, WaterServer = 3, UmbrellaSupplier = 4, Thief = 5 };
        [DataMember]
        public short GameId { get; set; }
        [DataMember]
        public short ProcessId { get; set; }
        [DataMember]
        public PossibleProcessType ProcessType { get; set; }
        [DataMember]
        public short LifePoints { get; set; }
        [DataMember]
        public short HitPoints { get; set; }
        [DataMember]
        public short NumberOfPennies { get; set; }
        [DataMember]
        public short NumberOfUnfilledBalloon { get; set; }
        [DataMember]
        public short NumberOfFilledBalloon { get; set; }
        [DataMember]
        public short NumberOfUnraisedUmbrellas { get; set; }
        [DataMember]
        public bool HasUmbrellaRaised { get; set; }
    }
}
